sgtm/dioscuri/dioscuri_get_gemini_response.tpl

```diff
@@ -0,0 +1,653 @@
+        "simpleValueType": true,
+        "help": "Specifies a list of strings that tells the model to stop generating text if one of the strings is encountered in the response. If a string appears multiple times in the response, then the response truncates where it\u0027s first encountered. The strings are case-sensitive.",
+        "valueValidators": [
+          {
+            "type": "REGEX",
+            "args": [
+              "[a-z]+(?:, +?[a-z]+){0,4}"
+            ],
+            "errorMessage": "Must be a comma-separated list containing a maximum of five words"
+          }
+        ]
+      }
+    ]
+  }
+]
+
+
+___SANDBOXED_JS_FOR_SERVER___
+
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/**
+ * @fileoverview sGTM variable tag that makes a call to Gemini API
+ * @version 1.0.0
+ */
+const getGoogleAuth = require("getGoogleAuth");
+const JSON = require("JSON");
+const logToConsole = require("logToConsole");
+const sendHttpRequest = require("sendHttpRequest");
+
+//Function to convert string of words to array of strings
+function splitAndTrim(commaSeparatedString) {
+  commaSeparatedString = "hello, goodbye";
+  const words = commaSeparatedString.split(',');
+  const trimmedWords = []; 
+  for (let i = 0; i < words.length; i++) {
+    trimmedWords.push(words[i].trim()); // Trim and add to new array
+  }
+  return trimmedWords;
+}
+
+// Build the URL for Vertex AI.
+const url = "https://" + data.cloudLocation + 
+  "-aiplatform.googleapis.com/v1/projects/" + data.projectID + 
+  "/locations/" + data.cloudLocation + "/publishers/google/models/" + 
+  data.geminiModel + ":generateContent";
+
+logToConsole(url);
+
+// Get Google credentials from the service account running the container.
+const auth = getGoogleAuth({
+  scopes: ["https://www.googleapis.com/auth/cloud-platform"]
+});
+
+// Create parts object which contains text and file prompts
+logToConsole(data);
+const parts = [];
+
+if (data.textPromptData) {
+  data.textPromptData.forEach((textPrompt) => {
+    parts.push({"text": textPrompt.text});
+  });
+}
+
+
+if (data.filePromptData && data.geminiModel == "gemini-1.5-pro") {
+  data.filePromptData.forEach((filePrompt) => {
+    parts.push({"fileData": {
+      "fileUri": filePrompt.fileUrl,
+      "mimeType": filePrompt.mimeType
+    }});
+  });
+}
+
+if (parts.length == 0) {
+  logToConsole("At least one prompt must be supplied");
+  //data.gtmOnFailure();
```

Comment:
Why is this commented out. Seems reasonable to fail here.

Reply:
Worked out that data.gtmOnFailure() is not available in variables which is when you get a not defined error. Removed